mlir/utils/jenkins/Jenkinsfile

```diff
@@ -170,180 +288,118 @@ pipeline {
-                                          -DROCMLIR_DRIVER_RANDOM_DATA_SEED=1
-                                          -DROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED=0
-                                          -DROCMLIR_DRIVER_TEST_GPU_VALIDATION=0
-                                          -DLLVM_LIT_ARGS='-v --time-tests'
-                                          -DCMAKE_EXPORT_COMPILE_COMMANDS=1
-                                        """)
-                            }
+                        steps {
+                            check_fixedE2ETests(params.nightly)
                         }
                     }
-                    post {
-                        unsuccessful {
-                            rebootNode()
+                    stage("Shared Library: random E2E tests") {
+                        when {
+                            beforeAgent true;
+                            allOf {
+                                equals expected: true, actual: params.sharedLib;
+                                equals expected: true, actual: params.nightly
+                            }
                         }
-                        always {
-                            cleanWs()
+                        steps {
+                            check_randomE2ETests()
                         }
                     }
-                }
-                stage("Static Library PR tests") {
-                    when {
-                        beforeAgent true;
-                        allOf {
+                    stage("Static Library: test librockCompiler with MIOpen") {
+                        when {
+                            beforeAgent true;
                             equals expected: true, actual: params.staticLib;
                             equals expected: false, actual: params.nightly;
-                            equals expected: false, actual: params.weekly;
                         }
-                    }
-                    agent {
-                        docker {
-                            image dockerImage()
-                            args dockerArgs()
-                            label "mlir && ${params.canXdlops ? '(gfx908 || gfx90a)' : 'gfx906' }"
-                            alwaysPull true
+                        environment {
+                            // Make libMIOpen.so accessible to the test driver
+                            LD_LIBRARY_PATH="${WORKSPACE}/MIOpen/build/lib:$LD_LIBRARY_PATH:"
                         }
-                    }
-                    environment {
-                        PATH="/opt/rocm/llvm/bin:$PATH"
-                        HOME="${WORKSPACE}"
-                    }
-                    stages {
-                        stage('Environment') {
-                            steps {
-                                showEnv()
-                            }
+                        steps {
+                            sh 'rm build/CMakeCache.txt'
```

Comment:
A thing I'd like you to test, though maybe in a followup PR:
Instead of deleting the CMake cache to force a full rebuild, edit the top-level CMakeLists.txt to go to that `if (BUILD_FAT_LIBROCKCOMPILER)` and set all the `set()`s in the arms of that if to include the `FORCE` flag so that we re-set the various variables we're controlling. This may also involve moving the `set()` of the install prefix out of the `if()`

Reply:
I'll address this issue in a future PR.